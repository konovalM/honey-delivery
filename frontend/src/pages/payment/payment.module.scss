@use '@styles/colors.scss' as *;

.container {
    padding: 2rem 0;
}

.mainTitle {
    text-align: center;
    color: $text-base;
    margin-bottom: 1.5rem !important;
}

.introText {
    text-align: center;
    font-size: 1.1rem;
    max-width: 800px;
    margin: 0 auto 2rem !important;
    color: #555;
}

.divider {
    margin: 2rem 0 !important;
    font-size: 1.2rem !important;
    color: #3498db !important;
}

.paymentMethods {
    display: grid;
    grid-template-columns: repeat(auto-fit, minmax(350px, 1fr));
    gap: 2rem;
    margin-bottom: 3rem;
}

.paymentCard {
    border-radius: 12px !important;
    overflow: hidden;
    box-shadow: 0 4px 12px rgba(0, 0, 0, 0.08);
    transition:
        transform 0.3s ease,
        box-shadow 0.3s ease;

    &:hover {
        transform: translateY(-5px);
        box-shadow: 0 6px 16px rgba(0, 0, 0, 0.12);
    }
}

.imageContainer {
    height: 200px;
    overflow: hidden;
    display: flex;
    align-items: center;
    justify-content: center;
    background: #f7f7f7;
}

.paymentImage {
    width: 100%;
    height: 100%;
    object-fit: cover;
}

.cardTitle {
    color: #2c3e50 !important;
    margin-bottom: 1rem !important;
}

.advantagesDivider,
.noteDivider {
    margin: 1rem 0 !important;
}

.advantagesList {
    width: 100%;
}

.advantageItem {
    display: flex;
    align-items: center;
    font-size: 1rem;
}

.noteText {
    font-size: 0.9rem !important;
    display: block;
}

.securityInfo {
    background: #f8f9fa;
    padding: 1.5rem;
    border-radius: 8px;
    margin-top: 2rem;
    border-left: 4px solid #3498db;
}

.securityTitle {
    color: #3498db !important;
    margin-bottom: 1rem !important;
}

@media (max-width: 768px) {
    .paymentMethods {
        grid-template-columns: 1fr;
    }

    .mainTitle {
        font-size: 1.8rem !important;
    }

    .introText {
        font-size: 1rem;
    }
}
