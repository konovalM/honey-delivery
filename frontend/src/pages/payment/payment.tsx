import React from 'react';
import { Card, Divider, Image, Space, Typography } from 'antd';
// import paymentCash from './images/payment-cash.jpg'; // Замените на свои изображения
// import paymentCard from './images/payment-card.jpg';
// import paymentOnline from './images/payment-online.jpg';
import cls from './payment.module.scss';

const { Title, Paragraph, Text } = Typography;

export const Payment = () => {
    const paymentMethods = [
        {
            id: 1,
            title: "Онлайн-оплата",
            image: 'https://img.freepik.com/premium-vector/online-payment-concept_118813-2685.jpg',
            description: "Безопасная оплата банковской картой через платежный шлюз Сбербанка",
            advantages: [
                "Мгновенное подтверждение оплаты",
                "Защита данных по стандарту PCI DSS",
                "Возможность оплаты картами Visa, Mastercard, МИР"
            ],
            note: "После подтверждения заказа вы будете перенаправлены на страницу оплаты"
        },
        {
            id: 2,
            title: "Оплата курьеру",
            image: 'https://img.freepik.com/premium-vector/cash-payment-concepts-style-illustration-eps-10-file_357500-4162.jpg',
            description: "Наличными или картой при получении заказа",
            advantages: [
                "Бесплатная доставка при заказе от 3000 ₽",
                "Возможность проверить товар перед оплатой",
                "Курьер имеет POS-терминал для оплаты картой"
            ],
            note: "Принимаем наличные и бесконтактные платежи по терминалу"
        },
        {
            id: 3,
            title: "Оплата в пункте выдачи",
            image: 'https://static.vecteezy.com/system/resources/previews/000/684/243/non_2x/online-payment.jpg',
            description: "В наших фирменных пунктах выдачи заказов",
            advantages: [
                "Более 10 ПВЗ в Краснодаре и СПБ",
                "Возможность примерки и проверки",
                "Оплата картой или наличными"
            ],
            note: "Адреса и график работы можно уточнить при оформлении заказа"
        }
    ];

    return (
        <div className={cls.container}>
            <Title level={1} className={cls.mainTitle}>Способы оплаты</Title>
            <Paragraph className={cls.introText}>
                Мы предлагаем удобные и безопасные способы оплаты для наших клиентов.
                Выберите наиболее подходящий вариант при оформлении заказа.
            </Paragraph>

            <Divider orientation="left" className={cls.divider}>
                <Text strong>Доступные методы</Text>
            </Divider>

            <div className={cls.paymentMethods}>
                {paymentMethods.map(method => (
                    <Card
                        key={method.id}
                        className={cls.paymentCard}
                        cover={
                            <div className={cls.imageContainer}>
                                <Image
                                    src={method.image}
                                    alt={method.title}
                                    preview={false}
                                    className={cls.paymentImage}
                                />
                            </div>
                        }
                    >
                        <Title level={3} className={cls.cardTitle}>{method.title}</Title>
                        <Paragraph>{method.description}</Paragraph>

                        <Divider dashed className={cls.advantagesDivider} />

                        <Space direction="vertical" size="small" className={cls.advantagesList}>
                            {method.advantages.map((item, index) => (
                                <Text key={index} className={cls.advantageItem}>
                                    ✓ {item}
                                </Text>
                            ))}
                        </Space>

                        <Divider dashed className={cls.noteDivider} />

                        <Text type="secondary" className={cls.noteText}>
                            {method.note}
                        </Text>
                    </Card>
                ))}
            </div>

            <div className={cls.securityInfo}>
                <Title level={4} className={cls.securityTitle}>Безопасность платежей</Title>
                <Paragraph>
                    Все платежи защищены современными технологиями шифрования.
                    Мы не храним данные вашей банковской карты на своих серверах.
                </Paragraph>
            </div>
        </div>
    );
};