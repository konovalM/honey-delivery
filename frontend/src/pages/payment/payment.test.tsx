import { render } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import { Payment } from './payment';

describe('Payment Page', () => {
  it('рендерится без ошибок и соответствует snapshot', () => {
    const { container } = render(<Payment />);
    expect(container).toMatchSnapshot();
  });
});
