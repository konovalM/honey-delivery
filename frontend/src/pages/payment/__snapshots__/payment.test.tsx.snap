// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Payment Page > рендерится без ошибок и соответствует snapshot 1`] = `
<div>
  <div
    class="_container_a7043b"
  >
    <h1
      class="ant-typography _mainTitle_a7043b css-dev-only-do-not-override-1d4w9r2"
    >
      Способы оплаты
    </h1>
    <div
      class="ant-typography _introText_a7043b css-dev-only-do-not-override-1d4w9r2"
    >
      Мы предлагаем удобные и безопасные способы оплаты для наших клиентов. Выберите наиболее подходящий вариант при оформлении заказа.
    </div>
    <div
      class="ant-divider css-dev-only-do-not-override-1d4w9r2 ant-divider-horizontal ant-divider-with-text ant-divider-with-text-start _divider_a7043b"
      role="separator"
    >
      <span
        class="ant-divider-inner-text"
      >
        <span
          class="ant-typography css-dev-only-do-not-override-1d4w9r2"
        >
          <strong>
            Доступные методы
          </strong>
        </span>
      </span>
    </div>
    <div
      class="_paymentMethods_a7043b"
    >
      <div
        class="ant-card ant-card-bordered _paymentCard_a7043b css-dev-only-do-not-override-1d4w9r2"
      >
        <div
          class="ant-card-cover"
        >
          <div
            class="_imageContainer_a7043b"
          >
            <div
              class="ant-image css-dev-only-do-not-override-1d4w9r2"
            >
              <img
                alt="Онлайн-оплата"
                class="ant-image-img _paymentImage_a7043b css-dev-only-do-not-override-1d4w9r2"
                src="https://img.freepik.com/premium-vector/online-payment-concept_118813-2685.jpg"
              />
            </div>
          </div>
        </div>
        <div
          class="ant-card-body"
        >
          <h3
            class="ant-typography _cardTitle_a7043b css-dev-only-do-not-override-1d4w9r2"
          >
            Онлайн-оплата
          </h3>
          <div
            class="ant-typography css-dev-only-do-not-override-1d4w9r2"
          >
            Безопасная оплата банковской картой через платежный шлюз Сбербанка
          </div>
          <div
            class="ant-divider css-dev-only-do-not-override-1d4w9r2 ant-divider-horizontal ant-divider-dashed _advantagesDivider_a7043b"
            role="separator"
          />
          <div
            class="ant-space css-dev-only-do-not-override-1d4w9r2 ant-space-vertical ant-space-gap-row-small ant-space-gap-col-small _advantagesList_a7043b"
          >
            <div
              class="ant-space-item"
            >
              <span
                class="ant-typography _advantageItem_a7043b css-dev-only-do-not-override-1d4w9r2"
              >
                ✓ 
                Мгновенное подтверждение оплаты
              </span>
            </div>
            <div
              class="ant-space-item"
            >
              <span
                class="ant-typography _advantageItem_a7043b css-dev-only-do-not-override-1d4w9r2"
              >
                ✓ 
                Защита данных по стандарту PCI DSS
              </span>
            </div>
            <div
              class="ant-space-item"
            >
              <span
                class="ant-typography _advantageItem_a7043b css-dev-only-do-not-override-1d4w9r2"
              >
                ✓ 
                Возможность оплаты картами Visa, Mastercard, МИР
              </span>
            </div>
          </div>
          <div
            class="ant-divider css-dev-only-do-not-override-1d4w9r2 ant-divider-horizontal ant-divider-dashed _noteDivider_a7043b"
            role="separator"
          />
          <span
            class="ant-typography ant-typography-secondary _noteText_a7043b css-dev-only-do-not-override-1d4w9r2"
          >
            После подтверждения заказа вы будете перенаправлены на страницу оплаты
          </span>
        </div>
      </div>
      <div
        class="ant-card ant-card-bordered _paymentCard_a7043b css-dev-only-do-not-override-1d4w9r2"
      >
        <div
          class="ant-card-cover"
        >
          <div
            class="_imageContainer_a7043b"
          >
            <div
              class="ant-image css-dev-only-do-not-override-1d4w9r2"
            >
              <img
                alt="Оплата курьеру"
                class="ant-image-img _paymentImage_a7043b css-dev-only-do-not-override-1d4w9r2"
                src="https://img.freepik.com/premium-vector/cash-payment-concepts-style-illustration-eps-10-file_357500-4162.jpg"
              />
            </div>
          </div>
        </div>
        <div
          class="ant-card-body"
        >
          <h3
            class="ant-typography _cardTitle_a7043b css-dev-only-do-not-override-1d4w9r2"
          >
            Оплата курьеру
          </h3>
          <div
            class="ant-typography css-dev-only-do-not-override-1d4w9r2"
          >
            Наличными или картой при получении заказа
          </div>
          <div
            class="ant-divider css-dev-only-do-not-override-1d4w9r2 ant-divider-horizontal ant-divider-dashed _advantagesDivider_a7043b"
            role="separator"
          />
          <div
            class="ant-space css-dev-only-do-not-override-1d4w9r2 ant-space-vertical ant-space-gap-row-small ant-space-gap-col-small _advantagesList_a7043b"
          >
            <div
              class="ant-space-item"
            >
              <span
                class="ant-typography _advantageItem_a7043b css-dev-only-do-not-override-1d4w9r2"
              >
                ✓ 
                Бесплатная доставка при заказе от 3000 ₽
              </span>
            </div>
            <div
              class="ant-space-item"
            >
              <span
                class="ant-typography _advantageItem_a7043b css-dev-only-do-not-override-1d4w9r2"
              >
                ✓ 
                Возможность проверить товар перед оплатой
              </span>
            </div>
            <div
              class="ant-space-item"
            >
              <span
                class="ant-typography _advantageItem_a7043b css-dev-only-do-not-override-1d4w9r2"
              >
                ✓ 
                Курьер имеет POS-терминал для оплаты картой
              </span>
            </div>
          </div>
          <div
            class="ant-divider css-dev-only-do-not-override-1d4w9r2 ant-divider-horizontal ant-divider-dashed _noteDivider_a7043b"
            role="separator"
          />
          <span
            class="ant-typography ant-typography-secondary _noteText_a7043b css-dev-only-do-not-override-1d4w9r2"
          >
            Принимаем наличные и бесконтактные платежи по терминалу
          </span>
        </div>
      </div>
      <div
        class="ant-card ant-card-bordered _paymentCard_a7043b css-dev-only-do-not-override-1d4w9r2"
      >
        <div
          class="ant-card-cover"
        >
          <div
            class="_imageContainer_a7043b"
          >
            <div
              class="ant-image css-dev-only-do-not-override-1d4w9r2"
            >
              <img
                alt="Оплата в пункте выдачи"
                class="ant-image-img _paymentImage_a7043b css-dev-only-do-not-override-1d4w9r2"
                src="https://static.vecteezy.com/system/resources/previews/000/684/243/non_2x/online-payment.jpg"
              />
            </div>
          </div>
        </div>
        <div
          class="ant-card-body"
        >
          <h3
            class="ant-typography _cardTitle_a7043b css-dev-only-do-not-override-1d4w9r2"
          >
            Оплата в пункте выдачи
          </h3>
          <div
            class="ant-typography css-dev-only-do-not-override-1d4w9r2"
          >
            В наших фирменных пунктах выдачи заказов
          </div>
          <div
            class="ant-divider css-dev-only-do-not-override-1d4w9r2 ant-divider-horizontal ant-divider-dashed _advantagesDivider_a7043b"
            role="separator"
          />
          <div
            class="ant-space css-dev-only-do-not-override-1d4w9r2 ant-space-vertical ant-space-gap-row-small ant-space-gap-col-small _advantagesList_a7043b"
          >
            <div
              class="ant-space-item"
            >
              <span
                class="ant-typography _advantageItem_a7043b css-dev-only-do-not-override-1d4w9r2"
              >
                ✓ 
                Более 10 ПВЗ в Краснодаре и СПБ
              </span>
            </div>
            <div
              class="ant-space-item"
            >
              <span
                class="ant-typography _advantageItem_a7043b css-dev-only-do-not-override-1d4w9r2"
              >
                ✓ 
                Возможность примерки и проверки
              </span>
            </div>
            <div
              class="ant-space-item"
            >
              <span
                class="ant-typography _advantageItem_a7043b css-dev-only-do-not-override-1d4w9r2"
              >
                ✓ 
                Оплата картой или наличными
              </span>
            </div>
          </div>
          <div
            class="ant-divider css-dev-only-do-not-override-1d4w9r2 ant-divider-horizontal ant-divider-dashed _noteDivider_a7043b"
            role="separator"
          />
          <span
            class="ant-typography ant-typography-secondary _noteText_a7043b css-dev-only-do-not-override-1d4w9r2"
          >
            Адреса и график работы можно уточнить при оформлении заказа
          </span>
        </div>
      </div>
    </div>
    <div
      class="_securityInfo_a7043b"
    >
      <h4
        class="ant-typography _securityTitle_a7043b css-dev-only-do-not-override-1d4w9r2"
      >
        Безопасность платежей
      </h4>
      <div
        class="ant-typography css-dev-only-do-not-override-1d4w9r2"
      >
        Все платежи защищены современными технологиями шифрования. Мы не храним данные вашей банковской карты на своих серверах.
      </div>
    </div>
  </div>
</div>
`;
